specs/packagemanager/PackageManagement.md

```diff
@@ -0,0 +1,1136 @@
+  references are routinely handled.
+* `PackageManager.AddPackageByUriAsync(p)` fails returning `ERROR_PACKAGE_ALREADY_EXISTS` if that
+  exact version of the package is already installed. `PackageDeploymentManager` succeeds as the
+  requested package is installed.
+* Many `PackageManager` operations accept a target package as a file but require it expressed as a
+  `Uri`. `PackageDeploymentManager` provides overrides also accepting it as a `String`.
+* `PackageManager.RemovePackageByFullNameAsyn(p)` fails if the specified package isn't found.
+  `PackageDeploymentManager` succeeds as the requested package is not present at the end of the
+  operation.
+  * This follows the core deployment principle "'Tis not the journey that matters but the
+    destination". In short, it doesn't matter the current state of the system, only the desired end
+    state is achieved.
+* `PackageManager` methods accept inconsistent (and often inconvenient) permutations of expressing a
+  target package. For example, `PackageManager` supports removing a package by PackageFullName but
+  not PackageFamilyName. `PackageDeploymentManager` provides a richer API accepting additional
+  identifiers.
+
+# 4. Examples
+
+## 4.1. AddPackageAsync()
+
+Fabrikam app installing Contoso's Muffin package from a .msix file.
+
+```c#
+void Install()
+{
+    var package = "d:\\contoso\\muffin.msix";
+    var packageDeploymentManager = PackageDeploymentManager.GetDefault();
+    var options = new AddPackageOptions();
+    var deploymentResult = await packageDeploymentManager.AddPackageAsync(package, options);
+    try
+    {
+        if (deplymentResult.Status == PackageDeploymentStatus.CompletedSuccess)
+        {
+            Console.WriteLine("OK");
+        }
+        else
+        {
+            Console.WriteLine("Error {}", deploymentResult.ExtendedError());
+        }
+    }
+    catch (Exception ex)
+    {
+        Console.WriteLine(ex.ToString());
+    }
+}
+```
+
+## 4.2. AddPackageByUriAsync()
+
+Fabrikam app installing Contoso's Muffin package from an https: source.
+
+```c#
+void Install()
+{
+    var package = new Uri("https://contoson.com/muffin.msix");
+    var packageDeploymentManager = PackageDeploymentManager.GetDefault();
+    var options = new AddPackageOptions();
+    var deploymentResult = await packageDeploymentManager.AddPackageByUriAsync(package, options);
+    try
+    {
+        if (deplymentResult.Status == PackageDeploymentStatus.CompletedSuccess)
+        {
+            Console.WriteLine("OK");
+        }
+        else
+        {
+            Console.WriteLine("Error {}", deploymentResult.ExtendedError());
+        }
+    }
+    catch (Exception ex)
+    {
+        Console.WriteLine(ex.ToString());
+    }
+}
+```
+
+**NOTE:** This differs from the AddPackageAsync() example by the method name
+(`AddPackageByUriAsync()` instead of `AddPackageAsync()`) and the target type (https: source as URI
+instead of filename as string).
```

Comment:
Why is the string version needed?  One can just as easily do
            var package = new Uri("d:\\contoso\\muffin.msix");
instead of
            var package = "d:\\contoso\\muffin.msix";

Reply:
Not everything can be expressed as a URI

* YES: file:, http(s):, ms-uup, path, filename
* NO: PackageFamilyName, PackageFullName

Plus "it's easy" in C#, not necessarily so easy in all cases (irksome friction in C++ ABI)